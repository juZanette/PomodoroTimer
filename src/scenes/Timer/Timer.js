import React from 'react'
import Box from '@material-ui/core/Box'
import { CountdownCircle } from './components/CountdownCircle'
import { ToggleButton } from './components/ToggleButton'
import { RoundsCounter } from './components/RoundsCounter'
import { ResetButton } from './components/ResetButton'
import { SkipButton } from './components/SkipButton'

const Timer = () => {
  return (
    <Box width={300} m="auto">
      <CountdownCircle />
      <ToggleButton />

      <Box
        display="flex"
        justifyContent="space-between"
        alignItems="center"
        mt={2}
      >
        <RoundsCounter />

        <Box display="flex">
          <ResetButton />
          <SkipButton />
        </Box>
      </Box>
    </Box>
  )
}

export { Timer }
