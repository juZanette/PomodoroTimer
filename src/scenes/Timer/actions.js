export const setStatus = status => ({
  type: 'SET_STATUS',
  status,
})

export const setProgress = progress => ({
  type: 'SET_PROGRESS',
  progress,
})

export const setTimeLeft = timeLeft => ({
  type: 'SET_TIME_LEFT',
  timeLeft,
})

export const saveInterval = interval => ({
  type: 'SAVE_INTERVAL',
  interval,
})

export const pauseTimer = () => ({
  type: 'PAUSE_TIMER',
})

export const resetTimer = () => ({
  type: 'RESET_TIMER',
})

export const setNextTimer = () => ({
  type: 'SET_NEXT_TIMER',
})
