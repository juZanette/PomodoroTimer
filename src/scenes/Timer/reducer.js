const initialState = {
  timerActive: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_TIMER_ACTIVE':
      return {
        timerActive: action.timerActive,
      }
    default:
      return state
  }
}
