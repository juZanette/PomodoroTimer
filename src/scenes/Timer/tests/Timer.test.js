import React from 'react'
import { createShallow } from '@material-ui/core/test-utils'
import { Timer } from '../Timer'

describe('<SkipButton />', () => {
  let shallow
  let wrapper

  beforeEach(() => {
    shallow = createShallow()
    wrapper = shallow(<Timer />)
  })

  test('should render <Timer /> correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
