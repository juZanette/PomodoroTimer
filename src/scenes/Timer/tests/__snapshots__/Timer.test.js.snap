// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipButton /> should render <Timer /> correctly 1`] = `
<Styled(MuiBox)
  m="auto"
  pt={3}
  width={300}
>
  <LabelButton />
  <CountdownCircle />
  <ToggleButton />
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    justifyContent="space-between"
    mt={2}
  >
    <RoundsCounter />
    <Styled(MuiBox)
      display="flex"
    >
      <ResetButton />
      <SkipButton />
    </Styled(MuiBox)>
  </Styled(MuiBox)>
  <FullscreenDialog />
  <DesktopDialog />
</Styled(MuiBox)>
`;
